test/test_transforms_v2_refactored.py

```diff
@@ -2208,9 +2180,96 @@ def new_resize(dp, *args, **kwargs):
         t(torch.rand(3, 10, 10)).shape == (3, 224, 224)
         t(datapoints.Image(torch.rand(3, 10, 10))).shape == (3, 224, 224)
 
-    def test_bad_disaptcher_name(self):
-        class CustomDatapoint(datapoints.Datapoint):
-            pass
+    def test_errors(self, mocker):
+        mocker.patch.dict(_KERNEL_REGISTRY, clear=True)
 
         with pytest.raises(ValueError, match="Could not find dispatcher with name"):
-            F.register_kernel("bad_name", CustomDatapoint)
+            F.register_kernel("bad_name", datapoints.Image)
+
+        with pytest.raises(ValueError, match="Kernels can only be registered on dispatchers"):
+            F.register_kernel(datapoints.Image, F.resize)
+
+        with pytest.raises(ValueError, match="Kernels can only be registered for subclasses"):
+            F.register_kernel(F.resize, object)
+
+        F.register_kernel(F.resize, datapoints.Image)(F.resize_image_tensor)
+
+        with pytest.raises(ValueError, match="already has a kernel registered for type"):
+            F.register_kernel(F.resize, datapoints.Image)(F.resize_image_tensor)
+
+
+class TestGetKernel:
+    def make_and_register_kernel(self, dispatcher, input_type):
+        return _register_kernel_internal(dispatcher, input_type, datapoint_wrapper=False)(object())
+
+    @pytest.fixture
+    def dispatcher_and_kernels(self, mocker):
+        mocker.patch.dict(_KERNEL_REGISTRY, clear=True)
+
+        dispatcher = object()
```

Comment:
honestly, I'd rather use an existing dispatcher like `F.resize()` rather than this `object()`. It may seem clear to you that there is no side effect and that we're indeed testing the logic we want to test, but I don't think it will be obvious to anyone else (it's really not clear to me, I'm just trusting that the test works as intended, but I really don't know).